core/src/main/java/org/elasticsearch/Version.java

```diff
@@ -196,7 +196,7 @@ public static Version fromString(String version) {
         if (snapshot = version.endsWith("-SNAPSHOT")) {
             version = version.substring(0, version.length() - 9);
         }
-        String[] parts = version.split("\\.|\\-");
+        String[] parts = version.split("[.-]");
```

Comment:
The performance difference here probably doesn't matter, although the new version is more readable.